runtime/runtime-params-estimator/src/function_call.rs

```diff
@@ -0,0 +1,150 @@
+use crate::cases::ratio_to_gas_signed;
+use crate::testbed_runners::{end_count, start_count, GasMetric};
+use crate::vm_estimator::{create_context, least_squares_method};
+use near_primitives::config::VMConfig;
+use near_primitives::contract::ContractCode;
+use near_primitives::profile::ProfileData;
+use near_primitives::runtime::fees::RuntimeFeesConfig;
+use near_primitives::types::{CompiledContractCache, ProtocolVersion};
+use near_store::{create_store, StoreCompiledContractCache};
+use near_vm_logic::mocks::mock_external::MockedExternal;
+use near_vm_runner::{run_vm, VMKind};
+use nearcore::get_store_path;
+use std::fmt::Write;
+use std::sync::Arc;
+
+#[allow(dead_code)]
+fn test_function_call(metric: GasMetric, vm_kind: VMKind) {
+    let mut xs = vec![];
+    let mut ys = vec![];
+    const REPEATS: i32 = 50;
+    for method_count in vec![5, 20, 30, 50, 100, 200, 1000] {
+        let contract = make_many_methods_contact(method_count);
+        let cost =
+            compute_function_call_cost(metric, vm_kind, REPEATS, &contract);
+        println!("{:?} {:?} {} {}", vm_kind, metric, method_count, cost / (REPEATS as u64));
+        xs.push(contract.get_code().len() as u64);
+        ys.push(cost / (REPEATS as u64));
+    }
+
+    // Regression analysis only makes sense for additive metrics.
+    if metric == GasMetric::Time {
+        return;
+    }
+
+    let (cost_base, cost_byte, _) = least_squares_method(&xs, &ys);
+
+    println!(
+        "{:?} {:?} function call base {} gas, per byte {} gas",
+        vm_kind, metric,
+        ratio_to_gas_signed(metric, cost_base),
+        ratio_to_gas_signed(metric, cost_byte),
+    );
+}
+
+#[test]
+fn test_function_call_time() {
+    // Run with
+    // cargo test --release --lib function_call::test_function_call_time -- --exact --nocapture
```

Comment:
Hm, shouldn't this include `--features required` as well?

I *think* as written, this'll hit the in-memory cache, masking the deserializaion costs, and desiralization should be the primary contributor to func call cost

Reply:
Done, although from results it doesn't seem so.